File: src/j/implementation/microSet/InnerSet_10.java
package j.implementation.microSet;

import java.util.Iterator;

public class InnerSet_10 <T> extends AbstractInnerSet<T>
{
        protected T element_1;
        protected T element_2;
        protected T element_3;
        protected T element_4;
        protected T element_5;
        protected T element_6;
        protected T element_7;
        protected T element_8;
        protected T element_9;
        protected T element_10;
        
    public InnerSet_10(T p_element_1, T p_element_2, T p_element_3, T p_element_4, T p_element_5, T p_element_6, T p_element_7, T p_element_8, T p_element_9, T p_element_10)
    {
                element_1 = p_element_1;
                element_2 = p_element_2;
                element_3 = p_element_3;
                element_4 = p_element_4;
                element_5 = p_element_5;
                element_6 = p_element_6;
                element_7 = p_element_7;
                element_8 = p_element_8;
                element_9 = p_element_9;
                element_10 = p_element_10;
            }

    public InnerSet_10(InnerSet_10<T> i)
    {
            element_1 = i.element_1;
            element_2 = i.element_2;
            element_3 = i.element_3;
            element_4 = i.element_4;
            element_5 = i.element_5;
            element_6 = i.element_6;
            element_7 = i.element_7;
            element_8 = i.element_8;
            element_9 = i.element_9;
            element_10 = i.element_10;
        }

    public InnerSet_10<T> copy() {
        return new InnerSet_10(this);
    }

    public InnerSet_10(InnerSet_9<T> i, T el)
    {
            element_1 = i.element_1;            element_2 = i.element_2;            element_3 = i.element_3;            element_4 = i.element_4;            element_5 = i.element_5;            element_6 = i.element_6;            element_7 = i.element_7;            element_8 = i.element_8;            element_9 = i.element_9;                    element_10 = el;
    }
    
    @Override
    public InnerSet<T> addElement(T p_element)
    {
        if(element_1.equals(p_element) || element_2.equals(p_element) || element_3.equals(p_element) || element_4.equals(p_element) || element_5.equals(p_element) || element_6.equals(p_element) || element_7.equals(p_element) || element_8.equals(p_element) || element_9.equals(p_element) || element_10.equals(p_element))
            return this;
        else
            return new InnerArraySet<T>(this, p_element);    }

    @Override
    public InnerSet<T> addUnChecked(T p_element)
    {
        return new InnerArraySet<T>(this, p_element);    }

    @Override
    public InnerSet<T> removeElement(Object p_element)
    {
                if(element_1.equals(p_element))
            return  new InnerSet_9<>(element_2, element_3, element_4, element_5, element_6, element_7, element_8, element_9, element_10);
         else                 if(element_2.equals(p_element))
            return  new InnerSet_9<>(element_1, element_3, element_4, element_5, element_6, element_7, element_8, element_9, element_10);
         else                 if(element_3.equals(p_element))
            return  new InnerSet_9<>(element_1, element_2, element_4, element_5, element_6, element_7, element_8, element_9, element_10);
         else                 if(element_4.equals(p_element))
            return  new InnerSet_9<>(element_1, element_2, element_3, element_5, element_6, element_7, element_8, element_9, element_10);
         else                 if(element_5.equals(p_element))
            return  new InnerSet_9<>(element_1, element_2, element_3, element_4, element_6, element_7, element_8, element_9, element_10);
         else                 if(element_6.equals(p_element))
            return  new InnerSet_9<>(element_1, element_2, element_3, element_4, element_5, element_7, element_8, element_9, element_10);
         else                 if(element_7.equals(p_element))
            return  new InnerSet_9<>(element_1, element_2, element_3, element_4, element_5, element_6, element_8, element_9, element_10);
         else                 if(element_8.equals(p_element))
            return  new InnerSet_9<>(element_1, element_2, element_3, element_4, element_5, element_6, element_7, element_9, element_10);
         else                 if(element_9.equals(p_element))
            return  new InnerSet_9<>(element_1, element_2, element_3, element_4, element_5, element_6, element_7, element_8, element_10);
         else                 if(element_10.equals(p_element))
            return  new InnerSet_9<>(element_1, element_2, element_3, element_4, element_5, element_6, element_7, element_8, element_9);
                        else
            return this;
    }

    @Override
    public boolean containsElement(Object p_element)
    {
        return element_1.equals(p_element) || element_2.equals(p_element) || element_3.equals(p_element) || element_4.equals(p_element) || element_5.equals(p_element) || element_6.equals(p_element) || element_7.equals(p_element) || element_8.equals(p_element) || element_9.equals(p_element) || element_10.equals(p_element);
    }


    public T getElement(int index)
    {switch(index) {
                case 1:
            return element_1;
                case 2:
            return element_2;
                case 3:
            return element_3;
                case 4:
            return element_4;
                case 5:
            return element_5;
                case 6:
            return element_6;
                case 7:
            return element_7;
                case 8:
            return element_8;
                case 9:
            return element_9;
                case 10:
            return element_10;
                default:
            return null;
            }
    }


    public InnerSet<T> addAllElements(InnerSet<T> innerSet) {
        return innerSet.addElement(element_1).addElement(element_2).addElement(element_3).addElement(element_4).addElement(element_5).addElement(element_6).addElement(element_7).addElement(element_8).addElement(element_9).addElement(element_10);
    }


    @Override
    public InnerSet<T> removeAllElements(InnerSet<T> innerSet) {
        return innerSet.removeElement(element_1).removeElement(element_2).removeElement(element_3).removeElement(element_4).removeElement(element_5).removeElement(element_6).removeElement(element_7).removeElement(element_8).removeElement(element_9).removeElement(element_10);
    }

    @Override
    public InnerSet<T> retainAllElements(InnerSet<T> innerSet) {
        InnerSet<T> i = this;
                if (!innerSet.containsElement(element_1)) i = i.removeElement(element_1);
                if (!innerSet.containsElement(element_2)) i = i.removeElement(element_2);
                if (!innerSet.containsElement(element_3)) i = i.removeElement(element_3);
                if (!innerSet.containsElement(element_4)) i = i.removeElement(element_4);
                if (!innerSet.containsElement(element_5)) i = i.removeElement(element_5);
                if (!innerSet.containsElement(element_6)) i = i.removeElement(element_6);
                if (!innerSet.containsElement(element_7)) i = i.removeElement(element_7);
                if (!innerSet.containsElement(element_8)) i = i.removeElement(element_8);
                if (!innerSet.containsElement(element_9)) i = i.removeElement(element_9);
                if (!innerSet.containsElement(element_10)) i = i.removeElement(element_10);
                return i;
    }

    @Override
    public boolean containsAllElements(InnerSet<T> innerSet) {
        return  innerSet.containsElement(element_1) && innerSet.containsElement(element_2) && innerSet.containsElement(element_3) && innerSet.containsElement(element_4) && innerSet.containsElement(element_5) && innerSet.containsElement(element_6) && innerSet.containsElement(element_7) && innerSet.containsElement(element_8) && innerSet.containsElement(element_9) && innerSet.containsElement(element_10);
    }

    public InnerSet<T> addAllAndPropagate(InnerSet<T> innerSet, MicroSet<T> microSet) {
        return innerSet.addAndPropagate(element_1, microSet).addAndPropagate(element_2, microSet).addAndPropagate(element_3, microSet).addAndPropagate(element_4, microSet).addAndPropagate(element_5, microSet).addAndPropagate(element_6, microSet).addAndPropagate(element_7, microSet).addAndPropagate(element_8, microSet).addAndPropagate(element_9, microSet).addAndPropagate(element_10, microSet);
    }

    public InnerSet<T> addAndPropagate(T p_element, MicroSet<T> microSet) {
        if(element_1.equals(p_element) || element_2.equals(p_element) || element_3.equals(p_element) || element_4.equals(p_element) || element_5.equals(p_element) || element_6.equals(p_element) || element_7.equals(p_element) || element_8.equals(p_element) || element_9.equals(p_element) || element_10.equals(p_element))
            return this;
        else {
            microSet.add(p_element);
            return new InnerArraySet<T>(this, p_element);        }
    }

    @Override
    public String toString()
    {
        return "{" + element_1 + ", " + element_2 + ", " + element_3 + ", " + element_4 + ", " + element_5 + ", " + element_6 + ", " + element_7 + ", " + element_8 + ", " + element_9 + ", " + element_10 + " }";
    }
    
    @Override
    public int getSize()
    {
        return 10;
    }

    @Override
    public Iterator<T> iterator() {
        return new InnerSetIterator<>(this);
    }

    @Override
    public InnerSet<T> clear(boolean unused) {
        return (InnerSet<T>) InnerSet_0.singleton;
    }
}

File: src/scala/implementation/InnerSet_10.scala
package scala.implementation


class InnerSet_10[T](protected val element1: T, protected val element2: T, protected val element3: T, protected val element4: T, protected val element5: T, protected val element6: T, protected val element7: T, protected val element8: T, protected val element9: T, protected val element10: T) extends AbstractInnerSet[T]
{

    def this(p_innerSet: InnerSet_10[T]) = this(p_innerSet.element1 , p_innerSet.element2 , p_innerSet.element3 , p_innerSet.element4 , p_innerSet.element5 , p_innerSet.element6 , p_innerSet.element7 , p_innerSet.element8 , p_innerSet.element9 , p_innerSet.element10 )
    override def addElement(p_element: T): InnerSet[T] =
    {
        if (element1.equals(p_element) || element2.equals(p_element) || element3.equals(p_element) || element4.equals(p_element) || element5.equals(p_element) || element6.equals(p_element) || element7.equals(p_element) || element8.equals(p_element) || element9.equals(p_element) || element10.equals(p_element))
        {
            this
        }
        else  new InnerSet_11(element1, element2, element3, element4, element5, element6, element7, element8, element9, element10, p_element)
    }

    override def addUnchecked(p_element: T): InnerSet[T] =  new InnerSet_11(element1, element2, element3, element4, element5, element6, element7, element8, element9, element10, p_element)

    override def addAllElements(p_innerSet: InnerSet[T]): InnerSet[T] =  p_innerSet.addElement(element1).addElement(element2).addElement(element3).addElement(element4).addElement(element5).addElement(element6).addElement(element7).addElement(element8).addElement(element9).addElement(element10)

    override def containsElement(p_element: Object): Boolean = element1.equals(p_element) || element2.equals(p_element) || element3.equals(p_element) || element4.equals(p_element) || element5.equals(p_element) || element6.equals(p_element) || element7.equals(p_element) || element8.equals(p_element) || element9.equals(p_element) || element10.equals(p_element)

    override def containsAllElements(p_innerSet: InnerSet[T]): Boolean =  p_innerSet.containsElement(element1.asInstanceOf[Object]) && p_innerSet.containsElement(element2.asInstanceOf[Object]) && p_innerSet.containsElement(element3.asInstanceOf[Object]) && p_innerSet.containsElement(element4.asInstanceOf[Object]) && p_innerSet.containsElement(element5.asInstanceOf[Object]) && p_innerSet.containsElement(element6.asInstanceOf[Object]) && p_innerSet.containsElement(element7.asInstanceOf[Object]) && p_innerSet.containsElement(element8.asInstanceOf[Object]) && p_innerSet.containsElement(element9.asInstanceOf[Object]) && p_innerSet.containsElement(element10.asInstanceOf[Object])

    override def getElement(p_index: Int): Option[T] = p_index match{
        case 1 => Some(element1)
        case 2 => Some(element2)
        case 3 => Some(element3)
        case 4 => Some(element4)
        case 5 => Some(element5)
        case 6 => Some(element6)
        case 7 => Some(element7)
        case 8 => Some(element8)
        case 9 => Some(element9)
        case 10 => Some(element10)
        case _ => None
    }


    override def removeElement(p_element: Object): InnerSet[T] =
    {
            if(element1.equals(p_element))
                if(element2.equals(p_element))
                if(element3.equals(p_element))
                if(element4.equals(p_element))
                if(element5.equals(p_element))
                if(element6.equals(p_element))
                if(element7.equals(p_element))
                if(element8.equals(p_element))
                if(element9.equals(p_element))
                if(element10.equals(p_element))
                 new InnerSet_9(element1, element2, element3, element4, element5, element6, element7, element8, element9)
                        else this
    }

    override def removeAllElements(p_innerSet: InnerSet[T]): InnerSet[T] =  p_innerSet.removeElement(element1.asInstanceOf[Object]).removeElement(element2.asInstanceOf[Object]).removeElement(element3.asInstanceOf[Object]).removeElement(element4.asInstanceOf[Object]).removeElement(element5.asInstanceOf[Object]).removeElement(element6.asInstanceOf[Object]).removeElement(element7.asInstanceOf[Object]).removeElement(element8.asInstanceOf[Object]).removeElement(element9.asInstanceOf[Object]).removeElement(element10.asInstanceOf[Object])

    override def retainAllElements(p_innerSet: InnerSet[T]): InnerSet[T] =
    {
      var i: InnerSet[T] = this
            if(!p_innerSet.containsElement(element1.asInstanceOf[Object])) i = i.removeElement(element1.asInstanceOf[Object])
            if(!p_innerSet.containsElement(element2.asInstanceOf[Object])) i = i.removeElement(element2.asInstanceOf[Object])
            if(!p_innerSet.containsElement(element3.asInstanceOf[Object])) i = i.removeElement(element3.asInstanceOf[Object])
            if(!p_innerSet.containsElement(element4.asInstanceOf[Object])) i = i.removeElement(element4.asInstanceOf[Object])
            if(!p_innerSet.containsElement(element5.asInstanceOf[Object])) i = i.removeElement(element5.asInstanceOf[Object])
            if(!p_innerSet.containsElement(element6.asInstanceOf[Object])) i = i.removeElement(element6.asInstanceOf[Object])
            if(!p_innerSet.containsElement(element7.asInstanceOf[Object])) i = i.removeElement(element7.asInstanceOf[Object])
            if(!p_innerSet.containsElement(element8.asInstanceOf[Object])) i = i.removeElement(element8.asInstanceOf[Object])
            if(!p_innerSet.containsElement(element9.asInstanceOf[Object])) i = i.removeElement(element9.asInstanceOf[Object])
            if(!p_innerSet.containsElement(element10.asInstanceOf[Object])) i = i.removeElement(element10.asInstanceOf[Object])
            i
    }

    override def getSize: Int = 10

    override def iterator: InnerSetIterator[T] = new InnerSetIterator(this)

    override def copy: InnerSet[T] = this

    override def clear(unused: Boolean): InnerSet[T] = InnerSet_0[T]

    override def toString : String = "{ " + element1 + ", " + element2 + ", " + element3 + ", " + element4 + ", " + element5 + ", " + element6 + ", " + element7 + ", " + element8 + ", " + element9 + ", " + element10 + " }"

    override def addAndPropagate(p_element: T, p_microSet: MicroSet[T]): InnerSet[T] =
    {
        if (element1.equals(p_element) || element2.equals(p_element) || element3.equals(p_element) || element4.equals(p_element) || element5.equals(p_element) || element6.equals(p_element) || element7.equals(p_element) || element8.equals(p_element) || element9.equals(p_element) || element10.equals(p_element))
        {
            this
        }
        else         {
            p_microSet.add(p_element)
            new InnerSet_11(element1, element2, element3, element4, element5, element6, element7, element8, element9, element10, p_element)
        }    }

    override def addAllAndPropagate(p_innerSet: InnerSet[T], p_microSet: MicroSet[T]): InnerSet[T] = p_innerSet.addAllAndPropagateReverse(this, p_microSet)

    override def addAllAndPropagateReverse(p_innerSet: InnerSet[T], p_microSet: MicroSet[T]): InnerSet[T] =  p_innerSet.addAndPropagate(element1,p_microSet).addAndPropagate(element2,p_microSet).addAndPropagate(element3,p_microSet).addAndPropagate(element4,p_microSet).addAndPropagate(element5,p_microSet).addAndPropagate(element6,p_microSet).addAndPropagate(element7,p_microSet).addAndPropagate(element8,p_microSet).addAndPropagate(element9,p_microSet).addAndPropagate(element10,p_microSet)
}